mgr: remove legacy mgr HA logic and watch-active sidecar (#11661)
rook/rook: Documentation/CRDs/Cluster/ceph-cluster-crd.md

For more details on the mons and when to choose a number other than `3`, see the [mon health doc](../../Storage-Configuration/Advanced/ceph-mon-health.md).
* `mgr`: manager top level section
    * `count`: set number of ceph managers between `1` to `2`. The default value is 2.

>     If there are two managers, all mgr services will point to the active mgr. Ceph standby manager pod stays

Reviewer: ```suggestion
    If there are two managers, all mgr services will point to the active mgr. Due to a readiness probe, the standby mgr pod stays
```

> 	in not-Ready state untill it becomes active (by the underlying ceph mgr HA mechanism).

Reviewer: ```suggestion
	in the `not-ready` state until Ceph internally changes it to active state.
```

Author: fixed
